provider/resources.go

```diff
@@ -371,6 +375,90 @@ func Provider() tfbridge.ProviderInfo {
 	return prov
 }
 
+func ReplaceImagePolicyImageRecursiveTypes(spec *schema.PackageSpec, rootType string) {
```

Comment:
This function doesn't actually have anything to do with `ImagePolicyImage` types. We should generalize the name.

```suggestion
func MakeRecursiveType(spec *schema.PackageSpec, rootType string) {
```